pkg/services/object/put/distributed.go

```diff
@@ -143,7 +149,22 @@ func (t *distributedTarget) Close() (oid.ID, error) {
 	t.objSharedMeta = object.EncodeReplicationMetaInfo(t.obj.GetContainerID(), t.obj.GetID(), t.obj.PayloadSize(), deletedObjs,
 		lockedObjs, t.obj.CreationEpoch(), t.networkMagicNumber)
 	id := t.obj.GetID()
-	return id, t.placementIterator.iterateNodesForObject(id, t.sendObject)
+	err := t.placementIterator.iterateNodesForObject(id, t.sendObject)
+	if err != nil {
+		return oid.ID{}, err
+	}
+
+	if t.localNodeInContainer {
+		t.metaMtx.RLock()
```

Comment:
there is no more concurrent access at this point, isn't it?

Reply:
that is right but still, mutex saves _exactly this field_ so lets read it safely too, if you don't mind. mb, it will save us in future changes